server/registry/names/common.go

```diff
@@ -85,3 +85,27 @@ const Location = "global"
 type Name interface {
 	String() string // all names have a string representation.
 }
+
+// ExportableName returns a name suitable for export.
+// Its project and location are removed, making it project-local,
+// and any revision ID is removed since revision ids cannot be imported.
+//
+// Note that currently revision ids are only removed from trailing segments.
+// That's all that we need now, but support for removing internal revision ids
+// could be added if needed.
+//
+// Also note: this would be a nice function to add to the Name interface
+// but that might require improving the regularity of access to ProjectIDs
+// for different types of Names.
+func ExportableName(name string, projectID string) string {
```

Comment:
I'm not sure I understand why this can't be put into the Name interface... and why do we need to pass in the projectID? A name is well-structured on it's own... so we should be able to easily identify and remove the `projects/x` and `locations/x` segments, no?

Reply:
> "so we should be able to easily identify and remove the projects/x and locations/x segments, no?"

Yes, true, I considered doing this but it seemed to go against the spirit of the `names` package, which I think is trying to encapsulate knowledge about name structure.

Reply:
#867

Reply:
> seemed to go against the spirit of the names package, which I think is trying to encapsulate knowledge about name structure

This explanation leaves me more confused than before, but I trust that you've given it thought.

Reply:
[Here](https://github.com/apigee/registry/commit/b289e729991d0b67448278bdcda7cfbe3b1a01b9) is @seaneganx's original PR creating the `names` package. A goal of that was to reduce the amount of pasting together of resource names that was scattered throughout the code. Looking back, I think it has definitely been helpful, but we've also been aware that it was incomplete (#446).